src/Microsoft.VisualStudio.Composition/ComposedPart.cs

```diff
@@ -195,9 +195,10 @@ private static string GetDiagnosticLocation(ExportDefinitionBinding export)
             {
                 return string.Format(
                     CultureInfo.CurrentCulture,
-                    "{0}.{1}",
+                    Strings.TypeNameWithAssemblyLocation,
                     export.PartDefinition.Type.FullName,
-                    export.ExportingMember.Name);
+                    export.ExportingMember.Name,
+                    export.PartDefinition.Type.GetTypeInfo().Assembly.GetName().Name);
```

Comment:
We'll likely want to know the full name of the assembly (e.g. the version number could help in diagnosing issues). Instead of using GetName().Name, I'd suggest using `Assembly.FullName` (which is basically just `Assembly.GetName().FullName`, I hope).